client/lib/user/user.d.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+import { Language } from '../../languages';
+import { GravatarOptions } from '../../gravatar/types';
+import { URL, JSONSerializable } from '../../types';
+
+type WPCOMError = { message: string };
+type WPCOMHeaders = { [ name: string ]: string } & { status: number };
+
+type WPCOMCallback = (
+	error: WPCOMError | null,
+	result: JSONSerializable | null,
+	headers: WPCOMHeaders
+) => void;
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: ( userId: number ) => void;
+	get: () => UserData;
+	fetch: () => Promise< void >;
+	handleFetchFailure: ( error: Error ) => void;
+	handleFetchSuccess: ( userdata: UserData ) => void;
+	getLanguage: () => Language | undefined;
+	getAvatarUrl: ( options: GravatarOptions ) => URL;
+	clear: () => Promise< void > | void;
+	sendVerificationEmail< Callback extends WPCOMCallback >( callback: Callback ): XMLHttpRequest;
+	sendVerificationEmail(): Promise< void >;
+	set: ( attributes: UserData ) => boolean;
+	decrementSiteCount: () => void;
+	incrementSiteCount: () => void;
+	verificationPollerCallback: ( signal: boolean ) => void;
+	checkVerification: () => void;
+	signalVerification: () => void;
+	dispatchToken: string;
+}
+
+export type UserData = { ID: number } & Partial< OptionalUserData >;
+
+export type OptionalUserData = {
+	abtests: unknown;
+	avatar_URL: string;
+	date: string;
+	description: string;
+	display_name: string;
+	email: string;
+	email_verified: boolean;
+	has_unseen_notes: boolean;
+	is_new_reader: boolean;
+	isRTL: boolean;
+	is_valid_google_apps_country: boolean;
+	localeSlug: string;
+	localeVariant: string;
+	logout_URL: string;
+	meta: unknown;
```

Comment:
`unknown` is an odd type to return in a declaration like this. it literally means we can't do anything with it so we might as well leave it out.

granted, we shouldn't leave it out. we can probably glean _some_ more information about it though to make it useful. the `list-store` users `meta.next_page` which this doesn't have here. the [developer console](https://developer.wordpress.com/docs/api/console/) shows only the RESTy `links` in `meta` for this call

<img width="509" alt="Screen Shot 2020-02-26 at 1 41 10 AM" src="https://user-images.githubusercontent.com/5431237/75327233-3ee81a00-5839-11ea-9be9-54cac9439a62.png">

we might not actually need it though we wouldn't want to drop it without being absolutely sure. looks safe to add the links though

```ts
meta: { links: Record<'self' | 'help' | 'site' | 'flags', URL> };
```

this way at least if someone is using another possible `meta` value it will flag as a type error and hopefully lead them here to enhance the definition.

Reply:
I've added that type, in my developer console, userData.meta also had the `data` property so I've added that as an optional property. I also gave abtests a `Record< string, string>` type, which matches what I saw in the dev console: `abtests: { abtest1: 'value1', ... }`. The last `unknown` was `social_login_connections`, I couldn't get example data for that tonight, I'd like to just leave that as unknown for now until someone comes to use it.